bundle/Resources/config/templating.yml

```diff
@@ -26,53 +17,64 @@ services:
+        class: eZ\Publish\Core\MVC\Legacy\Templating\Converter\GenericConverter
+        public: false
 
     ezpublish_legacy.templating.delegating_converter:
-        class: "%ezpublish_legacy.templating.delegating_converter.class%"
+        class: eZ\Publish\Core\MVC\Legacy\Templating\Converter\DelegatingConverter
+        public: false
         arguments: ["@ezpublish_legacy.templating.generic_converter"]
 
     ezpublish_legacy.templating.object_converter:
         alias: ezpublish_legacy.templating.delegating_converter
+        public: false
 
     ezpublish_legacy.templating.api_content_converter:
-        class: "%ezpublish_legacy.templating.api_content_converter.class%"
+        class: eZ\Publish\Core\MVC\Legacy\Templating\Converter\ApiContentConverter
+        public: false
         arguments: ["@ezpublish_legacy.kernel"]
         tags:
-            - {name: ezpublish_legacy.templating.converter, for: "%ezpublish.api.content.class%"}
-            - {name: ezpublish_legacy.templating.converter, for: "%ezpublish.api.location.class%"}
-            - {name: ezpublish_legacy.templating.converter, for: "%ezpublish.api.version.class%"}
+            - {name: ezpublish_legacy.templating.converter, for: eZ\Publish\Core\Repository\Values\Content\Content}
+            - {name: ezpublish_legacy.templating.converter, for: eZ\Publish\Core\Repository\Values\Content\Location}
+            - {name: ezpublish_legacy.templating.converter, for: eZ\Publish\Core\Repository\Values\Content\VersionInfo}
 
     ezpublish_legacy.templating.pageparts_converter:
-        class: "%ezpublish_legacy.templating.pageparts_converter.class%"
+        class: eZ\Publish\Core\MVC\Legacy\Templating\Converter\PagePartsConverter
+        public: false
         tags:
-            - {name: ezpublish_legacy.templating.converter, for: "%ezpublish.fieldType.ezpage.block.class%"}
-            - {name: ezpublish_legacy.templating.converter, for: "%ezpublish.fieldType.ezpage.zone.class%"}
+            - {name: ezpublish_legacy.templating.converter, for: eZ\Publish\Core\FieldType\Page\Parts\Block}
+            - {name: ezpublish_legacy.templating.converter, for: eZ\Publish\Core\FieldType\Page\Parts\Zone}
 
     ezpublish_legacy.templating.legacy_helper:
-        class: "%ezpublish_legacy.templating.legacy_helper.class%"
+        class: eZ\Publish\Core\MVC\Legacy\Templating\LegacyHelper
+        public: false
         arguments: ["@ezpublish_legacy.kernel"]
 
     ezpublish.templating.global_helper.legacy:
-        class: "%ezpublish.templating.global_helper.legacy.class%"
+        class: eZ\Publish\Core\MVC\Legacy\Templating\GlobalHelper
         parent: ezpublish.templating.global_helper.core
+        public: false
         calls:
             - [setLegacyHelper, ["@ezpublish_legacy.templating.legacy_helper"]]
 
     # Overriding core helper
     ezpublish.templating.global_helper:
         alias: ezpublish.templating.global_helper.legacy
+        public: false
```

Comment:
Is EzPublishCoreBundle\Controller deprecated?   The breaks it's 'getGlobalHelper' as it needs the service to be public or an exception is thrown and it is private now.

Reply:
My guess: helper should be adapted to inject the service instead of relying on contianer.

Reply:
I don't believe when aliasing a service we should change its viability either which is causing me the issue I see.

Reply:
I'm fine with it being public, no problem :)

Reply:
Another approach would be to use the [service locator feature](https://symfony.com/blog/new-in-symfony-3-3-service-locators) in `EzPublishCoreBundle\Controller` (Symfony 3.3+)

Reply:
See implementation in Symfony `AbstractController`: https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

Reply:
Hi guys, I've found this issue after upgrading from an old 5.4 to the new 2.5.6, solved by overriding service definition on my own services.yml file.

Reply:
Someone willing to do PR to drop this, I agree we should not change this from kernel.

Reply:
PR: https://github.com/ezsystems/LegacyBridge/pull/175